agent/command/git_test.go

```diff
@@ -461,8 +462,8 @@ func (s *GitGetProjectSuite) TestBuildHTTPCloneCommand() {
 	s.Require().Len(cmds, 5)
 	s.True(utility.ContainsOrderedSubset(cmds, []string{
 		"set +o xtrace",
-		"echo \"git clone https://[redacted github token]:x-oauth-basic@github.com/evergreen-ci/sample.git 'dir' --branch 'main'\"",
-		"git clone https://PROJECTTOKEN:x-oauth-basic@github.com/evergreen-ci/sample.git 'dir' --branch 'main'",
+		fmt.Sprintf("echo \"git clone https://%s:x-oauth-basic@github.com/evergreen-ci/sample.git 'dir' --branch 'main'\"", projectGitHubToken),
```

Comment:
Since these now return the unredacted commands, are there other tests that cover the functionality that `GetLoggerProducer` loggers redact sensitive secrets? Reading through some of the logic, it looks like the relies on some slightly risky assumptions, like the fact that the expansions must be passed as a pointer to the redaction opts so that the memory is shared between the task config and the redacting sender.

If not, we should either add new tests or make a separate ticket to add tests that the loggers do redact correctly because it's critical functionality at this point and we shouldn't rely on manual testing to verify that.

Reply:
I improved the test in this file ([here](https://github.com/evergreen-ci/evergreen/pull/8330/files#diff-ded36303449c76616129b25a1fd87e727ac08946b916989b25ec2571c428369bR325)) and I added another more specific test [here](https://github.com/evergreen-ci/evergreen/pull/8330/files#diff-7004e4811680a6c872d2ac5728038b5001714068ef99a941fc73ba150afa160eR32) to the base communicator.

The git test uses the mock communicator, so it isn't a full on test of how it would work in production.

The test on the base communicator is more of a test on how it will work on production.